src/embed_tests/TestPythonException.cs

```diff
@@ -86,9 +86,60 @@ public void TestPythonExceptionFormatNoTraceback()
             }
             catch (PythonException ex)
             {
-                // ImportError/ModuleNotFoundError do not have a traceback when not running in a script 
+                // ImportError/ModuleNotFoundError do not have a traceback when not running in a script
                 Assert.AreEqual(ex.StackTrace, ex.Format());
             }
         }
+
+        [Test]
+        public void TestPythonExceptionFormatNormalized()
+        {
+            try
+            {
+                PythonEngine.Exec("a=b\n");
+            }
+            catch (PythonException ex)
+            {
+                Assert.AreEqual("Traceback (most recent call last):\n  File \"<string>\", line 1, in <module>\nNameError: name 'b' is not defined\n", ex.Format());
+            }
+        }
+
+        [Test]
+        public void TestPythonException_PyErr_NormalizeException()
+        {
+            using (Py.GIL())
+            {
+                using (PythonEngine engine = new PythonEngine())
+                {
```

Comment:
`PythonEngine` has a constructor and a `Dispose` method? o-O

Actually, this might be the reason for the test failures. You should not need neither `Py.GIL`, nor `PythonEngine`. And I believe the tests fail because `PythonEngine.Dispose` shuts Python down, so it crashes trying to release GIL when `Py.GIL()` goes out of scope.